Examples/Fail/spinLockBrokenCAS.cvf

```diff
@@ -0,0 +1,40 @@
+/*
+ * Herlihy/Shavit-style compare and swap lock,
+ * with an intentionally broken compare and swap.
+ */
+
+shared bool lock;  // True iff the lock is taken.
+thread bool test;  // Used when trying to take the lock.
+
+/*
+ * Locks the CAS lock.
+ */
+method lock() {
+  {| emp |}
+    do {
+      {| emp |}
+        test = false;
+      {| if test == false then emp else False() |}
+        <| if (lock == test) { lock = true; } /* test = lock; */ |>
```

Comment:
This new example just uses the atomic conditionals syntax to set up an incomplete and buggy CAS.